lib/matplotlib/pyplot.py

```diff
@@ -232,18 +235,27 @@ def switch_backend(newbackend):
 
     rcParams['backend'] = rcParamsDefault['backend'] = newbackend
 
-    global _backend_mod, new_figure_manager, draw_if_interactive, _show
-    _backend_mod = backend_mod
-    new_figure_manager = Backend.new_figure_manager
-    draw_if_interactive = Backend.draw_if_interactive
-    _show = Backend.show
+    for func_name in ["new_figure_manager", "draw_if_interactive", "show"]:
+        globals()[func_name].__signature__ = inspect.signature(
+            getattr(_backend_mod, func_name))
 
     # Need to keep a global reference to the backend for compatibility reasons.
     # See https://github.com/matplotlib/matplotlib/issues/6092
     matplotlib.backends.backend = newbackend
 
 
-def show(*args, **kw):
+def new_figure_manager(*args, **kwargs):
```

Comment:
Is it correct that this signature gets overwritten by `switch_backend`? If so, we should document that in a comment. Same with `draw_if_interactive` and `show`.

Reply:
yes, done